core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/StackV4Endpoint.java

```diff
@@ -401,6 +401,13 @@ FlowIdentifier upgradeClusterByName(@PathParam("workspaceId") Long workspaceId,
     FlowIdentifier upgradeClusterByNameInternal(@PathParam("workspaceId") Long workspaceId, @PathParam("name") String name,
             @QueryParam("imageId") String imageId, @QueryParam("initiatorUserCrn") String initiatorUserCrn);
 
+    @PUT
+    @Path("internal/{crn}/upgrade_ccm")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Initiates the CCM tunnel type upgrade to the latest available version", nickname = "upgradeCcmByNameInternal")
+    FlowIdentifier upgradeCcmByNameInternal(@PathParam("name") String name,
+            @QueryParam("initiatorUserCrn") String initiatorUserCrn);
```

Comment:
Copying from my previous review, these are still unresolved:

* Add `@NotEmpty` (from `javax.validation.constraints`) to `name` and `initiatorUserCrn`.
* `initiatorUserCrn` should be annotated with `@ValidCrn(resource = CrnResourceDescriptor.USER)`.